pkg/cache/ttlCache.go

```diff
@@ -54,6 +55,10 @@ type entry struct {
 	expiration int64 // nanoseconds
 }
 
+// EvictionCallback is a function that will be called on entry eviction
+// from an ExpiringCache.
```

Comment:
Need some docs here.

What context is this callback invoked? Has the entry been removed from the cache already when the callback is invoked? Are there reentrancy concerns? Does the callback need to complete promptly since a lock is held?

Reply:
There should also be a unit test to demonstrate that the callback works properly.

Reply:
Added some more docs and a basic unit test. Thoughts?